packages/integration-karma/test/shadow-dom/Element-properties/Element.innerHTML.spec.js

```diff
@@ -38,16 +38,18 @@ describe('Element.innerHTML - set', () => {
         }).toThrowError(TypeError);
     });
 
-    it('should log an error when invoking setter for an element in the shadow', () => {
-        const elm = createElement('x-test', { is: Test });
-        document.body.appendChild(elm);
-
-        const div = elm.shadowRoot.querySelector('div');
-
-        expect(() => {
-            div.innerHTML = '<span>Hello World!</span>';
-        }).toLogErrorDev(
-            /\[LWC error\]: The `innerHTML` property is available only on elements that use the `lwc:dom="manual"` directive./
-        );
-    });
+    if (!process.env.NATIVE_SHADOW) {
```

Comment:
I don't think this is the appropriate approach to test this. With this change, the test only runs when synthetic shadow is enabled. 

Instead, we should test that the error message is only reported with synthetic shadow DOM.

```js
const invokeInnerHTML = () => div.innerHTML = '<span>Hello World!</span>';

if (process.env.NATIVE_SHADOW) {
  expect(invokeInnerHTML).not.toLogErrorDev();
} else {
  expect(invokeInnerHTML).toLogErrorDev(/\[LWC error\]: The `innerHTML` property is available only on elements that use the `lwc:dom="manual"` directive./);
}
```

Reply:
I updated this PR to only be about light DOM, because that's what the original bug was about, and @ekashida pointed out that we don't necessarily want to encourage component authors to deliberately break compat with synthetic shadow.